torch/distributed/_composable/fsdp/_fsdp_state.py

```diff
@@ -36,6 +36,17 @@ def __init__(self):
         self.is_last_backward: bool = True
 
 
+def disable_if_config_true(func):
+    @functools.wraps(func)
+    def fsdp_hook_wrapper(*args, **kwargs):
+        if torch._dynamo.config.skip_fsdp_hooks:
+            return torch._dynamo.disable(func, recursive=True)(*args, **kwargs)
+        else:
+            return func(*args, **kwargs)
+
+    return fsdp_hook_wrapper
```

Comment:
I think this can be simplified as 

```
if config...:
	return torch._dynamo.disable(func)
else:
        return func
```

No need of inner wrapper.

Reply:
we discussed about this: we are using ``@functools.wraps(func)`` and ``torch._dynamo.disable(func, recursive=True)(*args, **kwargs)`` because we want a chance to set `_dynamo.config` before calling the function.